core/build.gradle.kts

```diff
@@ -15,22 +15,36 @@ kotlin {
     android {
         publishAllLibraryVariants()
     }
+    macosX64()
 
     sourceSets {
         val commonMain by getting {
             dependencies {
-                api(coroutines("core"))
+                api(coroutines("core", version = "1.4.1-native-mt"))
```

Comment:
The [`native-mt` is for Native **multithreading** support](https://github.com/Kotlin/kotlinx.coroutines/issues/462). The multithreading support in `Couroutines` is needed in Kable since values are being passed between threads via `Channel`s and `Flow`s (e.g. when we receive values from Mac OS's Core Bluetooth callback for a characteristic change, it is called on a dedicated dispatcher, then we pass it to a `Channel` so it can be picked up on the Coroutine context / Dispatcher being used on the `Flow` to observe the characteristic changes).

From what I understand, the Kotlin team is working on a new memory model for Native, which when available, Coroutines will adopt and we'll get the multithread support baked into Coroutines (and will no longer need a dedicated branch / release version of Coroutines), until then, we'll need to use the specialized Coroutines version.

Reply:
For details re: choice of version specification at the `common` source set level, see https://github.com/JuulLabs/kable/pull/8#discussion_r525545873 thread.